service/matching/task_reader.go

```diff
@@ -111,18 +115,39 @@ func (tr *taskReader) Signal() {
 	}
 }
 
+func (tr *taskReader) updateBacklogAge(task *internalTask) {
+	if task.event.Data.CreateTime == nil {
+		return // should not happen but for safety
+	}
+	ts := timestamp.TimeValue(task.event.Data.CreateTime).UnixNano()
+	// update the backlogAge value
+	tr.backlogHeadCreateTime.Store(ts)
```

Comment:
it's confusing to say "the backlogAge value" above something that touches `backlogHeadCreateTime` instead. This is important to fix imo